lib/static-cell/src/lib.rs

```diff
@@ -28,23 +28,34 @@ impl<T> StaticCell<T> {
         }
     }
 
-    /// Gets mutable access to the contents of `self`.
+    /// Attempts to get mutable access to the contents of `self`.
     ///
     /// If a `StaticRef` for `self` still exists anywhere in the program, this
-    /// will panic.
-    pub fn borrow_mut(&self) -> StaticRef<'_, T> {
+    /// will return `None`.
+    pub fn try_borrow_mut(&self) -> Option<StaticRef<'_, T>> {
         let already_borrowed =
             AtomicBoolExt::swap(&self.borrowed, true, Ordering::Acquire);
         if already_borrowed {
-            panic!();
+            return None;
         }
         // Safety: the check above ensures that we are not producing an aliasing
         // &mut to our contents.
-        unsafe {
-            StaticRef {
-                contents: &mut *self.cell.get(),
-                borrow: &self.borrowed,
-            }
+        let contents = unsafe { &mut *self.cell.get() };
+        Some(StaticRef {
+            contents,
+            borrow: &self.borrowed,
+        })
+    }
+
+    /// Gets mutable access to the contents of `self`.
+    ///
+    /// If a `StaticRef` for `self` still exists anywhere in the program, this
+    /// will panic.
+    pub fn borrow_mut(&self) -> StaticRef<'_, T> {
+        if let Some(borrow) = self.try_borrow_mut() {
+            borrow
+        } else {
+            panic!();
         }
```

Comment:
```suggestion
        self.try_borrow_mut().unwrap()
```
?

Reply:
Ah, yeah, I guess `Option::unwrap` doesn't try to format anything, so we can use it without needing to add a dependency on `userlib` just to get `unwrap_lite`...